proofs.go

```diff
@@ -439,28 +475,48 @@ func GenerateCandidates(
 	challengeCount uint64,
 	privateSectorInfo SortedPrivateSectorInfo,
 ) ([]PoStCandidateWithTicket, error) {
-	filReplicas, filReplicasLen, err := toFilPrivateReplicaInfos(privateSectorInfo.Values())
+	panic("unsupported")
+}
+
+// GenerateWinningPoStSectorChallenge
+func GenerateWinningPoStSectorChallenge(
+	proofType abi.RegisteredProof,
+	minerID abi.ActorID,
+	randomness abi.PoStRandomness,
+	eligibleSectors []abi.SectorNumber,
```

Comment:
This will be a _very_ large list, ~33 million for a 1EiB miner, 264MB. A bitfield would be a much better representation here, saving a big cost to the caller in return for a modest cost of indexing (scanning) the bitfield to look up items.

Even better would be to defer the index->sectornumber lookup to the caller, saving generation of any list structure (the caller can look up an AMT).

Reply:
removed :)